src/common/containers/builds.js

```diff
@@ -89,29 +55,15 @@ Builds.propTypes = {
     fullName: PropTypes.string.isRequired,
     url: PropTypes.string.isRequired
   }).isRequired,
-  isFetching: PropTypes.bool,
-  snap: PropTypes.shape({
-    self_link: PropTypes.string.isRequired,
-    store_name: PropTypes.string.isRequired
-  }),
-  success: PropTypes.bool,
-  error: PropTypes.object,
-  dispatch: PropTypes.func.isRequired
-};
-
-const mapStateToProps = (state, ownProps) => {
-  const owner = ownProps.params.owner.toLowerCase();
-  const name = ownProps.params.name.toLowerCase();
-  const fullName = `${owner}/${name}`;
-  const repository = state.repository;
-  // get builds for given repo from the store or set default empty values
-  const repoBuilds = state.snapBuilds[fullName] || snapBuildsInitialStatus;
-
-  return {
-    fullName,
-    repository,
-    ...repoBuilds
-  };
+  snapBuilds: PropTypes.shape({
+    isFetching: PropTypes.bool,
+    snap: PropTypes.shape({
+      self_link: PropTypes.string.isRequired,
+      store_name: PropTypes.string.isRequired
+    }),
+    success: PropTypes.bool,
+    error: PropTypes.object
+  })
 };
 
-export default connect(mapStateToProps)(withRepository(Builds));
+export default withRepository(withSnapBuilds(Builds));
```

Comment:
I moved `connect` out from here to higher order components.

So now this `Builds` component doesn't need to know how data is mapped (from state/params, etc).
Before that this component needed to define `mapStateToProps` to prepare data for both `withRepository` and `withSnapBuilds` wrapping components, but now they do it on their own.

Little drawback here is that now wrapping components have built-in data binding, so it may be harder to reuse them in different context. But I'm not sure if that will be the case - for now they do their job well - they hide repeated functionality and provide data to components that can be purely visual and not care about fetching data.